src/evaluator.ts

```diff
@@ -176,7 +177,10 @@ export class Evaluator {
     return false;
   }
 
-  private evaluateClause(clause: Clause, target: Target): boolean {
+  private async evaluateClause(
```

Comment:
You probably want to `await` any async calls in the `switch` statement as well

Reply:
only this is promise
```
return this.isTargetIncludedOrExcludedInSegment(clause.values, target);
```